text/0090-tikv-gc.md

```diff
@@ -0,0 +1,188 @@
+        - Need to design new interfaces for update service safepoint with 'service group'.
+        - How to allocate $service_group_id ?
+   2. design the etcd path to save service safepoint of service group:  
+        - the safepoint data path in etcd of PD,will be changed. The new safe point path in etcd as follows:
+        - gc_worker safe point
+         ```shell
+         /gc_servicegroup/gc_safepoint/$service_group_id
+         ```
+         - CDC,BR service safepoint
+         ```shell
+         /gc_servicegroup/service_safepoint/$service_group_id/$service_id
+         ```
+   3. design interface to standardize the interface:  
+         the interface as follows:  
+         1. Gc worker will call pbclient.UpdateGCSafePointByServiceGroup to update the gc worker safepoint as follows:  
+            1. interface :  
+               ```proto
+                   rpc UpdateGCSafePointByServiceGroup(UpdateGCSafePointByServiceGroupRequest) returns (UpdateGCSafePointByServiceGroupResponse) {}
+               ```
+       
+            2. PB messages:
+               ```proto
+                    message UpdateGCSafePointByServiceGroupRequest {
+                        RequestHeader header = 1;
+                        bytes service_group_id = 2;
+                        uint64 safe_point = 3;
+                        int64 revision = 4;
+                    }
+
+                    message UpdateGCSafePointByServiceGroupResponse {
+                        ResponseHeader header = 1;
+                        uint64 new_safe_point = 2;
+                        bool valid_revision = 3;
+                    }
+               ```
+         2. used to get all GC safepoint for TiKV:
+            1. interface:
+               ```proto  
+                   rpc GetAllServiceGroupGCSafePoints(GetAllServiceGroupGCSafePointRequest) returns (GetAllServiceGroupGCSafePointsResponse) {}
+               ```
+            2. PB messages:
+               ```proto
+                    message GetAllServiceGroupGCSafePointsRequest {
+                        RequestHeader header = 1;
+                    }
+
+                    message GetAllServiceGroupGCSafePointsResponse {
+                        ResponseHeader header = 1;
+                        repeated ServiceGroupSafePoint service_group_safe_point = 2;
+                    }
+               ```
+         3. Used to update service safepoint for CDC/BR/Lightning:
+            1. interface:
+               ```proto  
+                   rpc UpdateServiceSafePointByServiceGroup(UpdateServiceSafePointByServiceGroupRequest) returns (UpdateServiceSafePointByServiceGroupResponse) {}
+               ```
+            2. PB messages:
+               ```proto
+                    message UpdateServiceSafePointByServiceGroupRequest {
+                        RequestHeader header = 1;
+                        bytes service_group_id = 2;
+                        bytes service_id = 3;
+                        int64 TTL = 4;
```

Comment:
TTL doesn't follow field name standard. If `ttl` generates strange field accessor, we can change it to xxx_ttl  or maybe time_to_live. They both look better and consistent.